compiler/rustc_passes/src/stability.rs

```diff
@@ -164,8 +167,24 @@ impl<'a, 'tcx> Annotator<'a, 'tcx> {
 
         let (stab, const_stab) = attr::find_stability(&self.tcx.sess, attrs, item_sp);
 
-        let const_stab = const_stab.map(|(const_stab, _)| {
+        let const_stab = const_stab.map(|(const_stab, const_span)| {
             let const_stab = self.tcx.intern_const_stability(const_stab);
+            match item_kind {
+                Some(ItemKind::Fn(ref fn_sig, _, _)) => {
```

Comment:
This won't catch associated functions or trait functions, which are [`ImplItemKind::Fn`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/enum.ImplItemKind.html#variant.Fn) and [`TraitItemKind::Fn`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/enum.TraitItemKind.html#variant.Fn) respectively. I don't know if `rustc_const_unstable` is ever used on trait functions, but it's definitely used on methods, we need to handle them.